### PR TITLE
Add udp drops

### PR DESCRIPTION
**Pull request by alebsys**
Changes add parsing of the field drops in /proc/net/udp{,6} files.
Fixes: #506

### CODE REVIEW

File: net_ip_socket.go [hunk 1]
```diff
@@ -24,6 +24,12 @@ import (
 	"strings"
 )
 
+var (
+	// isUDP is used to determine the type of network socket UDP or TCP.
+	// Further its value will be used to parse UDP-specific fields from /proc/net/udp{,6} files.
+	isUDP bool
```
Comment (discordianfish) [inline net_ip_socket.go:30]: Why is this a global variable?

File: net_ip_socket.go [hunk 2]
```diff
@@ -224,5 +248,14 @@ func parseNetIPSocketLine(fields []string) (*netIPSocketLine, error) {
 		return nil, fmt.Errorf("%s: Cannot parse inode value in %q: %w", ErrFileParse, line.Inode, err)
 	}
 
+	// drops
+	if isUDP {
+		drops, err := strconv.ParseUint(fields[12], 0, 64)
```
Comment (discordianfish) [inline net_ip_socket.go:253]: What is fields[12] for non udp? Would it make sense to always set drops if fields[12] exist?

Comment (alebsys) [reply net_ip_socket.go:253]: Unfortunately, after the inode column, the file structure is different - in the `/proc/net/udp` file it is drops, but in `/proc/net/tcp` it is [retransmit timeout](https://www.kernel.org/doc/Documentation/networking/proc_net_tcp.txt).

File: net_ip_socket.go (hunk 1 above)
Comment (alebsys) [reply net_ip_socket.go:30]: You are right, I will change to a local variable.

File: net_ip_socket.go (hunk 2 above)
Comment (discordianfish) [reply net_ip_socket.go:248]: Ok makes sense

**Review by discordianfish — APPROVED** [review]
LGTM

File: net_ip_socket.go [hunk 3]
```diff
@@ -77,13 +81,18 @@ func newNetIPSocket(file string) (NetIPSocket, error) {
 	defer f.Close()
 
 	var netIPSocket NetIPSocket
+	var isUDP bool // is used to determine the type of network socket UDP or TCP
+
+	if strings.Contains(file, "udp") {
+		isUDP = true
+	}
```
Comment (SuperQ) [inline net_ip_socket.go:88]:
```suggestion
	isUDP := strings.Contains(file, "udp")
```

File: net_ip_socket.go [hunk 4]
```diff
@@ -104,19 +113,29 @@ func newNetIPSocketSummary(file string) (*NetIPSocketSummary, error) {
 	defer f.Close()
 
 	var netIPSocketSummary NetIPSocketSummary
+	var udpPacketDrops uint64
+	var isUDP bool // is used to determine the type of network socket UDP6 or TCP6
+
+	if strings.Contains(file, "udp") {
+		isUDP = true
+	}
```
Comment (SuperQ) [inline net_ip_socket.go:121]:
```suggestion
	isUDP := strings.Contains(file, "udp")
```

**Review by SuperQ — CHANGES_REQUESTED** [review]
A nice simplification.

**Comment by discordianfish** [comment]
@SuperQ needs to approve/merge

**Comment by darkrift** [comment]
Any chance we can see this move forward ? @SuperQ

**Comment by darkrift** [comment]
@discordianfish Can you do something to get this move forward ?

**Review by SuperQ — APPROVED** [review]

**Pull request merged**